### PR TITLE
add union in response cases

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:120]: string or int-array is already in route, why need to have it in request body?

File: packages/cadl-ranch-specs/cadl-ranch-summary.md
Comment (archerzz) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:3570]:
```suggestion
### Type_Union_receiveFirstTypeOfNamedUnionValue
```

File: packages/cadl-ranch-specs/cadl-ranch-summary.md
Comment (archerzz) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:3606]:
```suggestion
### Type_Union_receiveSecondTypeOfNamedUnionValue
```

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (MaryGao) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:128]: If this is only testing the response could we change `post` to `get` and then remove the request body part?

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (MaryGao) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:146]: +1 as above

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp:128]: Is there a difference?
Get could have a response, Post could also have a response body.
I could remove the request body, that is fine.
I am fine with changing the verb to get but I wonder if there is a difference

Comment (MaryGao) [reply packages/cadl-ranch-specs/http/type/union/main.tsp:128]: I understand both the GET and POST could test the response body but I prefer GET without any request body with following reasons:
- the type info is already in route we don't need to duplicate them in request body
- downstream could focus the existing changing part to do verification in our case that is the response body and no need to care others

I think this is our convertion in cadl-ranch with minimal code to make it work, but feel free to correct me if not!
/cc @weidongxu-microsoft @tadelesh

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/union/main.tsp:128]: I'd agree with Mary.

My main concern was the request body. Since you are going to remove the request body, and you do not test a POST, a GET is a better solution (IMHO).

There may not be significant difference between POST and GET. But since you are going to modify it anyway, GET is the most simple one when you don't have reason to use other verb.

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp:120]: OK I could remove. And how to make two scenarios which one of them send the first type of union, the other send the other type of the union? Should I add another `@scenario` to this operation?

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp:118]: body removed and changed to get

File: packages/cadl-ranch-specs/cadl-ranch-summary.md
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:3606]: I would like the name to keep alignment to the existing cases: https://github.com/Azure/cadl-ranch/blob/a71bd94f358a3ca8e276e2345a87b023b05eda85/packages/cadl-ranch-specs/http/type/union/mockapi.ts#L20
Unless I could change them both otherwise I rather keep these names.

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:118]: get?

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:42]: no longer needed?

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:131]: get?

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/main.tsp:144]: get?

File: packages/cadl-ranch-specs/http/type/union/mockapi.ts
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/mockapi.ts:35]: same, "get" here in all

File: packages/cadl-ranch-specs/http/type/union/mockapi.ts
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/union/mockapi.ts:36]: I don't think you need to "expect body"

Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/mockapi.ts:36]: should I write `req.expect.bodyEquals(undefined)`?
Now I removed this

File: packages/cadl-ranch-specs/http/type/union/mockapi.ts
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/mockapi.ts]: fixed

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp]: fixed

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp]: fixed

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp]: Yes, removed

File: packages/cadl-ranch-specs/http/type/union/main.tsp
Comment (ArcturusZhang) [reply packages/cadl-ranch-specs/http/type/union/main.tsp]: fixed

**Comment by weidongxu-microsoft** [comment]
LGTM for Java.

Still need an approval from interpreted language.

CI need to pass before merge.

**Comment by ArcturusZhang** [comment]
> LGTM for Java.
> 
> Still need an approval from interpreted language.
> 
> CI need to pass before merge.

Thank you! CI should be fixed now

**Review by MaryGao — APPROVED** [review]
LGTM for JS.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**
